libraries/CAN_comms/README.md

```diff
@@ -0,0 +1,99 @@
+
+# **CAN Communications Layer**
+
+- [**CAN Communications Layer**](#can-communications-layer)
+  - [**Overview**](#overview)
+- [**Quick Start**](#quick-start)
+  - [**Prerequisites**](#prerequisites)
+  - [**Initialization**](#initialization)
+  - [**Usage**](#usage)
+
+## **Overview**
+
+The CAN Communications Layer is a library that abstractes and simplifies the process of sending and receiving CAN messages by encapsulating CAN handling in a user-friendly API. This library leverages the HAL_CAN library for low-level CAN operations and FreeRTOS for managing asynchronous tasks, message queues, and semaphores.
+
+# **Quick Start**
+
+## **Prerequisites**
+
+**STM32F1 series MCU**  
+
+- This library was developed and tested with the STM32F1 series of MCUs. Other similar STM MCUs may work if they use the same underlying HAL.
+
+**CAN Interrupts**  
+
+- In the .ioc file, navigate to **Pinout & Configuration** -> **Connectivity** -> **CAN**.
+- Check the **Activated** box.
+- Check the **USB high priority or CAN TX interrupts** and **USB low priority or CAN RX0 interrupts** boxes.
+
+**CMSIS_V2 FreeRTOS**  
+
+- In the .ioc file, navigate to **Pinout & Configuration** -> **Middleware and Software Packs** -> **FREERTOS**.
+- Select **CMSIS_V2** on the **Interface** drop down box.
+- CAN_comms uses around 1024 bytes of heap space. You may need to increase the total heap size for FreeRTOS.
+
+**Importing library**  
+
+- See the **Libraries** section on the [wiki](https://wiki.ubcsolar.com/tutorials/stm32cubeide) for how to import a library to a project.
+
+**Code Generator**  
+
+- In the .ioc file, navigate to **Project Manager** -> **Code Generator**.
+- Check the **Generate peripheral initialization as a pair of '.c/.h' files per peripheral** box.
+
+## **Initialization**
+
+**Include library**  
+
+- Include the library using `#include "CAN_comms.h"`
+
+**Create config struct**  
+
+- Create a `CAN_comms_config_t` struct.
+
+  ```c
+  CAN_comms_config_t CAN_comms_config = {
+      .hcan = &hcan,
+      .CAN_Filter = CAN_filter,
+      .CAN_comms_Rx_callback = CAN_comms_Rx_callback
+    };
+  ```
+
+- `hcan` should be a pointer to the `CAN_HandleTypeDef` struct that's normally defined in `can.h`
+- `CAN_Filter` should be a `CAN_FilterTypeDef` struct.
+- `CAN_comms_Rx_callback` should be a pointer to a function of the form `void (*CAN_comms_Rx_callback)(CAN_comms_Rx_msg_t*);` This function will be called whenever a CAN message is received and ready to be parsed.
```

Comment:
Mention where to define this function. If you remove can.h then defining this callback in can.h is probably the most reasonable spot.